lib/generators/muscat/templates/devise_saml_authenticatable.rb

```diff
@@ -0,0 +1,77 @@
+  after_action :store_winning_strategy, only: :create
+
+  private
+
+  def store_winning_strategy
+    warden.session(:user)[:strategy] = warden.winning_strategies[:user].class.name.demodulize.underscore.to_sym
+  end
+end
+
+Devise.saml_update_resource_hook= Proc.new do |user, saml_response, auth_value|
+  user.user_create_strategy= :saml_authenticatable
+  user.roles << Role.find_by(name: RISM::SAML_AUTHENTICATION_CREATE_USER_ROLE)
+  Devise.saml_default_update_resource_hook.call(user, saml_response, auth_value)
+end
+
+Devise.setup do |config|
+  # ==> Configuration for :saml_authenticatable
+
+  # Create user if the user does not exist. (Default is false)
+  config.saml_create_user = true
+
+  # Update the attributes of the user after a successful login. (Default is false)
+  config.saml_update_user = true
+
+  # Set the default user key. The user will be looked up by this key. Make
+  # sure that the Authentication Response includes the attribute.
+  config.saml_default_user_key = :email
+
+  # Optional. This stores the session index defined by the IDP during login.  If provided it will be used as a salt
+  # for the user's session to facilitate an IDP initiated logout request.
+  config.saml_session_index_key = :session_index
+
+  # You can set this value to use Subject or SAML assertation as info to which email will be compared.
+  # If you don't set it then email will be extracted from SAML assertation attributes.
+  # config.saml_use_subject = true
+
+  # You can support multiple IdPs by setting this value to the name of a class that implements a ::settings method
+  # which takes an IdP entity id as an argument and returns a hash of idp settings for the corresponding IdP.
+  # config.idp_settings_adapter = "MyIdPSettingsAdapter"
+
+  # You provide you own method to find the idp_entity_id in a SAML message in the case of multiple IdPs
+  # by setting this to the name of a custom reader class, or use the default.
+  # config.idp_entity_id_reader = "DeviseSamlAuthenticatable::DefaultIdpEntityIdReader"
+
+  # You can set a handler object that takes the response for a failed SAML request and the strategy,
+  # and implements a #handle method. This method can then redirect the user, return error messages, etc.
+  # config.saml_failed_callback = nil
+
+  # You can customize the named routes generated in case of named route collisions with
+  # other Devise modules or libraries. Set the saml_route_helper_prefix to a string that will
+  # be appended to the named route.
+  # If saml_route_helper_prefix = 'saml' then the new_user_session route becomes new_saml_user_session
+  config.saml_route_helper_prefix = 'saml'
+
+  # You can add allowance for clock drift between the sp and idp.
+  # This is a time in seconds.
+  # config.allowed_clock_drift_in_seconds = 0
+
+  # idp_metadata_parser = OneLogin::RubySaml::IdpMetadataParser.new
+  # config.saml_config= idp_metadata_parser.parse_remote "https://remote-sso-server.example.org/saml/idp/metadata"
```

Comment:
If I understand well the role of this file, as a template that generates the config/initializers/devise_saml_authenticata\
ble.rb, I think that this file should be without comment, and this way:

config.saml_config = idp_metadata_parser.parse_remote "https://remote-sso-server.example.org/cas/idp/metadata"

I would change /saml/ to /cas/ because we have tested it with CAS server, and at least it will provide a good template for us.

Reply:
CAS is the name of a specific product, but SAML is the actual standard.